sources/filesystem/readers.py

```diff
@@ -74,6 +77,53 @@ def _read_parquet(
                 yield rows.to_pylist()
 
 
+def _read_csv_duckdb(
+    items: Iterator[FileItemDict],
+    chunk_size: Optional[int] = 5000,
+    use_pyarrow: bool = False,
+    read_csv_kwargs: Optional[Dict[str, Any]] = None,
+) -> Iterator[TDataItems]:
+    """A resource to extract data from the given CSV files.
+
+    Uses DuckDB engine to import and cast CSV data.
+
+    Args:
+        items (Iterator[FileItemDict]): CSV files to read.
+        chunk_size (Optional[int]):
+            The number of rows to read at once. Defaults to 5000.
+        use_pyarrow (bool):
+            Whether to use `pyarrow` to read the data and designate
+            data schema. If set to False (by default), JSON is used.
+        read_csv_kwargs (Optional[Dict]):
+            Additional keyword arguments to pass to the `read_csv()`.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    import duckdb
+    import pendulum
+
+    read_csv_kwargs = read_csv_kwargs or {}
+
+    state = dlt.current.resource_state()
```

Comment:
you really do not need it here. you can enable this behavior on the `filesystem` resource from which this transformer takes data. so the filtering of already processed files happens before transformer is called. here's an example:
```
# here we modify filesystem resource so it will track only new csv files
    # such resource may be then combined with transformer doing further processing
    new_files = filesystem(bucket_url=TESTS_BUCKET_URL, file_glob="csv/*")
    # add incremental on modification time
    new_files.apply_hints(incremental=dlt.sources.incremental("modification_date"))
    load_info = pipeline.run((new_files | read_csv()).with_name("csv_files"))
```
btw. this behavior is disabled by default